### PR TITLE
Calypso - write secrets

### PR DESCRIPTION
**Pull request by pierluca**

### CODE REVIEW

File: contracts/calypso/calypso.go [hunk 1]
```diff
@@ -137,6 +163,11 @@ func (c Contract) Execute(snap store.Snapshot, step execution.Step) error {
 		if err != nil {
 			return xerrors.Errorf("failed to LIST_SMC: %v", err)
 		}
+	case CmdCreateSecret:
+		err := c.cmd.createSecret(snap, step)
+		if err != nil {
+			return xerrors.Errorf("failed to CREATE_SECRET: %v", err)
+		}
```
Comment (jbsv) [inline contracts/calypso/calypso.go:170]: 'case CmdListSecrets:' missing ?

**Review by jbsv — CHANGES_REQUESTED** [review]
is it possible to implement and test listSecrets at this stage ?

**Comment by pierluca** [comment]
@jbsv Sorry, I had most things in place and forgot to add tests for LIST_SECRETS. 🤦  
Thank you for pointing that issue out.

File: contracts/calypso/calypso_test.go [hunk 2]
```diff
@@ -145,6 +151,201 @@ func TestCommand_ListSmc(t *testing.T) {
...
+	require.True(t, found)
+
+	require.Equal(t, 1, len(contract.secrets["other"]))
+
+	// Act
+	err = cmd.listSecrets(snap, makeStep(t, KeyArg, "dummy"))
+
+	// Assert
+	require.NoError(t, err)
+	require.Equal(t, fmt.Sprintf("name1=secret1,name2=secret2"), buf.String())
```
Comment (jbsv) [inline contracts/calypso/calypso_test.go:297]: unnecessary use of fmt.Sprintf (S1039)

File: contracts/calypso/calypso.go [hunk 3]
```diff
@@ -56,8 +67,19 @@ const (
 	// CmdDeleteSmc defines a command to deleteSmc a SMC
```
Comment (jbsv) [inline contracts/calypso/calypso.go:67]:
```suggestion
	// CmdDeleteSmc defines a command to delete a SMC
```

**Comment by pierluca** [comment]
Closing this PR as the logic was moved to the HBT repository.
https://github.com/dedis/hbt/pull/53

**Pull request closed without merging**